mne/tests/test_source_estimate.py

```diff
@@ -1548,3 +1582,153 @@ def test_stc_near_sensors(tmpdir):
+            op.join(tempdir, subject_to, 'bem',
+                    f'{subject_to}-oct-4-src.fif'))
+        labels_to = read_labels_from_annot(
+            subject_to, subjects_dir=tempdir)
+        # Save time since we know these subjects are identical
+        monkeypatch.setattr(mne.surface, '_make_morph_map_hemi',
+                            _make_morph_map_hemi_same)
+    else:
+        src_to = read_source_spaces(
+            op.join(tempdir, subject_to, 'bem',
+                    f'{subject_to}-vol20-src.fif'))
+        labels_to = op.join(
+            tempdir, subject_to, 'mri', 'aseg.mgz')
+    # 1. Label->STC->Label for the given subject should be identity
+    #    (for surfaces at least; for volumes it's not as clean as this
+    #     due to interpolation)
+    n_times = 50
+    rng = np.random.RandomState(0)
+    label_tc = rng.randn(n_labels, n_times)
+    # check that a random permutation of our labels yields a terrible
+    # correlation
+    corr = np.corrcoef(label_tc.ravel(),
+                       rng.permutation(label_tc).ravel())[0, 1]
+    assert -0.06 < corr < 0.06
+    # project label activations to full source space
+    stc = labels_to_stc(labels_from, label_tc, src=src_from)
+    assert isinstance(stc, klass)
+    label_tc_from = extract_label_time_course(
+        stc, labels_from, src_from, mode='mean')
+    if kind == 'surface':
+        assert_allclose(label_tc, label_tc_from, rtol=1e-12, atol=1e-12)
+    else:
+        corr = np.corrcoef(label_tc.ravel(), label_tc_from.ravel())[0, 1]
+        assert 0.66 < corr < 0.69
+
+    #
+    # 2. Changing STC subject to the surrogate and then extracting
+    #
+    stc.subject = subject_to
+    label_tc_to = extract_label_time_course(
+        stc, labels_to, src_to, mode='mean')
+    assert_allclose(label_tc_from, label_tc_to, rtol=1e-12, atol=1e-12)
+    stc.subject = subject_from
+
+    #
+    # 3. Morphing STC to new subject then extracting
+    #
+    if isinstance(scale, tuple) and kind == 'volume':
+        ctx = pytest.raises(RuntimeError, match='Non-uniform zooms')
+        test_morph = False
+    elif kind == 'surface':
+        ctx = pytest.warns(RuntimeWarning, match='not included')
+        test_morph = True
+    else:
+        ctx = nullcontext()
+        test_morph = True
+    with ctx:  # vertices not included
+        morph = compute_source_morph(
+            src_from, subject_to=subject_to, src_to=src_to,
+            subjects_dir=tempdir, niter_sdr=(), smooth=1, verbose=True)
+    if test_morph:
+        stc_to = morph.apply(stc)
+        label_tc_to_morph = extract_label_time_course(
+            stc_to, labels_to, src_to, mode='mean')
+        if kind == 'volume':
+            corr = np.corrcoef(
+                label_tc.ravel(), label_tc_to_morph.ravel())[0, 1]
+            if scale == 1.:
+                assert 0.42 < corr <= 0.47, corr
+            else:
+                # XXX even just a pure scaling does not work very well, bug...
```

Comment:
This is #8453

Reply:
I have a fix for this locally, but to save you some reviewing pain @agramfort feel free to review and merge this enhancement first